core/src/main/java/io/aiven/kafka/tieredstorage/metrics/CaffeineStatsCounter.java

```diff
@@ -39,184 +40,180 @@ public class CaffeineStatsCounter implements StatsCounter {
-        cacheEvictionSensor.record();
-        evictionSensorByCause(cause).record();
-        cacheEvictionWeightSensor.record(weight);
-        evictionWeightSensorByCause(cause).record(weight);
-    }
-
-    Sensor evictionSensorByCause(final RemovalCause cause) {
-        return new SensorProvider(metrics, "cause." + cause.name() + "." + CACHE_EVICTION,
-            () -> Map.of("cause", cause.name()))
-            .with(metricCacheEvictionRateByCause, new Rate())
-            .with(metricCacheEvictionTotalByCause, new CumulativeSum())
-            .get();
-    }
-
-    Sensor evictionWeightSensorByCause(final RemovalCause cause) {
-        return new SensorProvider(metrics, "cause." + cause.name() + "." + CACHE_EVICTION_WEIGHT,
-            () -> Map.of("cause", cause.name()))
-            .with(metricCacheEvictionWeightByCause, new Rate())
-            .with(metricCacheEvictionWeightTotalByCause, new CumulativeSum())
-            .get();
+        cacheEvictionCountTotal.increment();
+        cacheEvictionCountByCause.get(cause).increment();
+        cacheEvictionWeightTotal.add(weight);
+        cacheEvictionWeightByCause.get(cause).add(weight);
     }
 
     @Override
     public CacheStats snapshot() {
         return CacheStats.of(
-            (long) metrics.metric(metrics.metricInstance(metricCacheHitsTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheMissesTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheLoadSuccessTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheLoadFailureTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheLoadSuccessTimeTotal)).metricValue()
-                + (long) metrics.metric(metrics.metricInstance(metricCacheLoadFailureTimeTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheEvictionTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheEvictionWeightTotal)).metricValue()
+            negativeToMaxValue(cacheHitCount.sum()),
+            negativeToMaxValue(cacheMissCount.sum()),
+            negativeToMaxValue(cacheLoadSuccessCount.sum()),
+            negativeToMaxValue(cacheLoadFailureCount.sum()),
+            negativeToMaxValue(cacheLoadSuccessTimeTotal.sum() + cacheLoadFailureTimeTotal.sum()),
+            negativeToMaxValue(cacheEvictionCountTotal.sum()),
+            negativeToMaxValue(cacheEvictionWeightTotal.sum())
         );
     }
 
     @Override
     public String toString() {
         return snapshot().toString();
     }
+
+    /**
+     * Not called by Caffeine directory. Used to record custom cache miss.
+     */
+    public void recordMiss() {
+        cacheMissCount.increment();
+    }
+
+    /**
+     * Not called by Caffeine directory. Used to record custom cache hit.
+     */
+    public void recordHit() {
+        cacheHitCount.increment();
```

Comment:
Are these used or planned to?

Reply:
Yeah, we can leave them for `DiskBasedCache` PR

Reply:
Fixed